scripts/pre_commit_linter.py

```diff
@@ -2489,6 +2489,57 @@ def _check_constants_declaration(self):
 
         return summary_messages
 
+    def _check_npm_packages(self):
+        """Checks to ensure that npm packages are not identified as vulnerable
+        by npm audit.
+        """
+        parent_dir = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+
+        npm_path = os.path.join(
+            parent_dir, 'oppia_tools', 'node-10.15.3', 'bin', 'npm')
+        if self.verbose_mode_enabled:
+            print 'Starting npm package vulnerability check...'
+            print '----------------------------------------'
+        print ''
+        if not self.verbose_mode_enabled:
+            print 'Checking npm packages.'
+
+        summary_messages = []
+        failed = False
+
+        proc_args = [npm_path, 'audit', '--parseable']
+        with _redirect_stdout(_TARGET_STDOUT):
+            proc = subprocess.Popen(
+                proc_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+            linter_stdout, _ = proc.communicate()
+
+            errors = linter_stdout.lstrip().rstrip()
+            if errors:
+                failed = True
+                error_list = errors.split('\n')
+                for line in error_list:
+                    line_tokens = line.split('\t')
+                    print ('%s is vulnerable. Please run "%s" to update.' %
+                           (line_tokens[1], line_tokens[3]))
+
+
+            if failed:
+                print '(%s errors found)' % len(error_list)
+                summary_message = '%s  npm package vulnerability failed' % (
+                    _MESSAGE_TYPE_FAILED)
+                summary_messages.append(summary_message)
```

Comment:
I think we can pull this line out (here and below).

Reply:
done!